### PR TITLE
[GCP] Add GCP Billing Metricset

### PR DESCRIPTION
**Pull request by legoguy1000**
## What does this PR do?

Add GCP Billing Metricset

## Checklist

- [X] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/master/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [X] I have verified that all data streams collect metrics or logs.
- [X] I have added an entry to my package's `changelog.yml` file.
- [X] If I'm introducing a new feature, I have modified the Kibana version constraint in my package's `manifest.yml` file to point to the latest Elastic stack release (e.g. `^7.13.0`).

## Related issues

- Relates #490

### CODE REVIEW

**Comment by legoguy1000** [comment]
Split from #2066

File: packages/gcp/changelog.yml [hunk 1]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "1.2.0"
+  changes:
+    - description: Add GCP Billing Metricset
+      type: enhancement
+      link: https://github.com/elastic/integrations/pull/XX
```
Comment (endorama) [inline packages/gcp/changelog.yml:6]: Adding correct PR link to changelog
```suggestion
      link: https://github.com/elastic/integrations/pull/2141
```

**Comment by endorama** [comment]
@legoguy1000 yes please! Thanks

**Comment by legoguy1000** [comment]
@endorama updated.  Ready for review/test.

**Comment by legoguy1000** [comment]
@endorama Are you good to merge this?

**Comment by endorama** [comment]
Hello @legoguy1000 I've been working on adding tests but is taking longer than expected. The PR works and tests are green, so I'm going to merge and continue working in the background.

Thank you for the contribution!

**Comment by legoguy1000** [comment]
@endorama do u want me to start building the other PRs for the other metric data sets?

**Comment by endorama** [comment]
Yes please.

**Pull request merged**